### PR TITLE
Code of conduct

### PR DESCRIPTION
**Pull request by castelao**
A work in progress for the code of conduct. I'm starting with the standard GitHub and adding some ideas from other sources.

### CODE REVIEW

**Review by soerenthomsen — COMMENTED** [review]
Thanks Gui! This is a great starting point. Appreciate your work!

**Comment by soerenthomsen** [comment]
Thanks @castelao, I wonder whether we should place the Code of Conduct in the OceanGlidersCommunity/OceanGliders repository and link it in the readme file there? What do you think?

**Comment by castelao** [comment]
Once it is done, it is probably a good idea to have it in the main
repository. This is a small and important file, so I would keep it on each
repository as well.

**Review by isgiddy — COMMENTED** [review]
This is a really amazing and comprehensive Code of Conduct. Thank you.
 I only suggest adding a link to the Citizen Code of Conduct so that people (like me) who haven't heard of it can read more. Perhaps also adding something about acknowledging the work of others? I know this is common practice but maybe mention here? This might link into the point you added about focusing on the community.

**Comment by ptestor** [comment]
I have read it and it is great! Thanks!

**Review by ptestor — APPROVED** [review]
approved

**Review by soerenthomsen — APPROVED** [review]

**Review by LouClement — APPROVED** [review]

**Review by justinbuck — APPROVED** [review]
This policy is well aligned to comparable policies such as AGU and makes for a great community to be a part of. I fully support these guidelines.

**Review by justinbuck — APPROVED** [review]
Looks like minor updates since i last reviewed it

**Comment by soerenthomsen** [comment]
We plan to start the community review for the oxygen SOP next week. 

@castelao do you think a first release of the Code of Conduct is possible until then? Or do you want more people to feedback on it before?

I would be glad to link it like I did [here](https://oceangliderscommunity.github.io/Oxygen_SOP/README.html) here. If you prefer to direct people to the discussion here fine too.

**Review by AnthonyBosse — APPROVED** [review]

**Comment by AnthonyBosse** [comment]
I support the idea of this code of conduct. My only comment is that it is now quite long, hence it take time to get the overview of it.

**Review by tomhull — APPROVED** [review]

**Comment by castelao** [comment]
> @castelao I agree with you that we should merge this soon and put it into force. One question I have is whether it is possible to make sure it is in every repo but kept synced in case changes are made? Another option is that we place it in the [OceanGliders](https://github.com/OceanGlidersCommunity/OceanGliders) repo and link in every README? not sure what is best.

@soerenthomsen , Another alternative is to place it in the [.github repo](https://github.com/OceanGlidersCommunity/.github). I'll work on this PR tomorrow. Let's finish this first then we figure out where to place it.

**Comment by castelao** [comment]
@isgiddyk, just checking. Are you still interested in having something explicit about acknowledgments? Would you like to suggest what/where to add to the text? Note that it could be also done later, in a future PR.

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+* Trolling, insulting or derogatory comments, and personal or political attacks
+* Public or private harassment
+* Publishing others' private information, such as a physical or email
+  address, without their explicit permission
+* Other conduct which could reasonably be considered inappropriate in a
+  professional setting
+
+## Enforcement Responsibilities
+
+Community leaders are responsible for clarifying and enforcing our standards of
```
[the remark this answers is not in the document]
Comment (castelao) [reply CODE_OF_CONDUCT.md:66]: @nizaroka , I agree with you. Would you like to [open an issue in our central repository](https://github.com/OceanGlidersCommunity/OceanGliders/issues) to initiate that discussion? How to select the members, how to rotate, ...?

File: CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+not aligned to this Code of Conduct, and will communicate reasons for moderation
+decisions when appropriate.
+
+## Scope
+
+This Code of Conduct applies within all community spaces, and also applies when
+an individual is officially representing the community in public spaces.
+Examples of representing our community include using an official e-mail address,
+posting via an official social media account, or acting as an appointed
+representative at an online or offline event.
```
[the remark this answers is not in the document]
Comment (castelao) [reply CODE_OF_CONDUCT.md:82]: @nizaroka , thanks.

Can we have non-online events that are not in-person? Probably a silly question, but offline might cover in concept more possibilities than in-person only.

I would leave like that. Anyone with a strong opinion about offline vs in-person?

File: CODE_OF_CONDUCT.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+nationality, personal appearance, race, religion, sexual identity and
+orientation, or other similar characteristic.
+
+We pledge to act and interact in ways that contribute to an open, friendly,
+safe, welcoming, diverse, inclusive, and healthy community.
+
+## Our Standards
+
+* Please avoid using overtly sexual aliases or other nicknames that might detract from a friendly, safe and welcoming environment for all.
+* Please be kind and courteous. There’s no need to be mean or rude.
```
[the remark this answers is not in the document]
Comment (castelao) [reply CODE_OF_CONDUCT.md:18]: @nizaroka , I think that any mistreatment would be considered mean or rude, no? Do we expand the coverage by adding mistreatment? Anyways, I would be happy to add it if you have a strong opinion about this.

Thanks!

File: CODE_OF_CONDUCT.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+* Respect that people have differences of opinion and that every design or implementation choice carries a trade-off and numerous costs. There is seldom a right answer.
+* Please keep unstructured critique to a minimum. If you have solid ideas you want to experiment with, make a fork and see how it works.
+* We will exclude you from interaction if you insult, demean or harass anyone. That is not welcome behavior. We interpret the term “harassment” as including the definition in the Citizen Code of Conduct; if you have any lack of clarity about what might be included in that concept, please read [their definition](#harassment). In particular, we don’t tolerate behavior that excludes people in socially marginalized groups.
```
[the remark this answers is not in the document]
Comment (castelao) [reply CODE_OF_CONDUCT.md]: Done. Thanks!

**Comment by soerenthomsen** [comment]
> @soerenthomsen , Another alternative is to place it in the [.github repo](https://github.com/OceanGlidersCommunity/.github). I'll work on this PR tomorrow. Let's finish this first then we figure out where to place it.

Sounds good to me. Also after migrating the main README to the .github repo feel free to remove the old repo

**Review by nizaroka — APPROVED** [review]

File: CODE_OF_CONDUCT.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (nizaroka) [reply CODE_OF_CONDUCT.md:18]: @castelao I see your point you are right! Thank you for clarifying for me!

**Review by nizaroka — APPROVED** [review]

**Review by nizaroka — APPROVED** [review]

File: CODE_OF_CONDUCT.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (castelao) [reply CODE_OF_CONDUCT.md:18]: Thank you!

**Comment by castelao** [comment]
Thank you everyone for the comments, suggestions, and support. Just the interest in the subject by itself tells a lot about this community.

I'll wait another couple of days and if we don't have new comments I'll merge this PR. Thank you all.

**Comment by isgiddy** [comment]
> @isgiddy, just checking. Are you still interested in having something explicit about acknowledgments? Would you like to suggest what/where to add to the text? Note that it could be also done later, in a future PR.

I'm happy for you to go ahead. I think attribution is better mentioned in the contributing guidelines. Thanks for your work on this!

**Review by isgiddy — APPROVED** [review]

**Comment by soerenthomsen** [comment]
@castelao I think we can go ahead and merge this ok? The discussion concerning a committee and/or person(s) of contact has been moved [here](https://github.com/OceanGlidersCommunity/OceanGliders/issues) by @nizaroka.

File: CODE_OF_CONDUCT.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (castelao) [reply CODE_OF_CONDUCT.md:82]: I'll close this. Anyone with concerns about this, please open a new issue. Thanks!

**Comment by castelao** [comment]
Thank you all for the support and comments. I'm moving this forward, but it will be always open for discussion by opening a new issue.

**Comment by soerenthomsen** [comment]
Great thanks! Were will we place the CODE_OF_CONDUCT.md file? As soon as it is migrated to the final place I will link it in all repos. @castelao

**Pull request merged**